### DISCUSSION TITLE
Classroom assigment deadline cutoff date

### DISCUSSION THREAD

**Discussion by rafaelcorsi** [community/community#67707] [GitHub Education]
### Hi everyone,

If the deadline is set to 'This is a cutoff date', then once the assignments reach this deadline, GitHub changes the students' permissions on the repo to 'read-only'. However, if the deadline is subsequently edited, GitHub doesn't revert the permissions back to 'write'.

**Comment by daniel-bogale** [comment]
To ensure that GitHub permissions are reverted to 'write' when the deadline is edited, you could implement a custom automation or script. Here's a suggestion on how you might achieve this:

GitHub Actions:
Utilize GitHub Actions, a powerful automation tool provided by GitHub, to monitor changes to the deadline and adjust permissions accordingly. Here's a general outline of the steps:

Set up a GitHub Actions workflow that triggers on changes to the deadline file or a specific tag/commit.
In the workflow, use the GitHub API to update the repository permissions and revert them to 'write' when the deadline is modified.
The GitHub Actions workflow can use a script or a custom program to interact with the GitHub API and make the necessary permission changes.
Custom Script:
Create a custom script that periodically checks the deadline and adjusts permissions as needed. This script could run on a server or in a continuous integration environment.

The script would use the GitHub API to monitor changes to the deadline.
When a change is detected, the script would modify the repository permissions accordingly.
By implementing one of these approaches, you can ensure that GitHub permissions are adjusted back to 'write' when the deadline is edited. Keep in mind that you'll need to handle authentication to access the GitHub API securely and ensure proper error handling in case of any issues.

**Reply by rafaelcorsi** [reply]
Or gh could fix this bug
